src/Core/Contact/Application/UseCase/FindContactGroups/FindContactGroups.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * Copyright 2005 - 2022 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Contact\Application\UseCase\FindContactGroups;
+
+use Centreon\Domain\Contact\Contact;
+use Centreon\Domain\Log\LoggerTrait;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Core\Application\Common\UseCase\ForbiddenResponse;
+use Core\Contact\Application\Repository\ReadContactGroupRepositoryInterface;
+
+class FindContactGroups
+{
+    use LoggerTrait;
+
+    /**
+     * @param ReadContactGroupRepositoryInterface $repository
+     * @param ContactInterface $user
+     */
+    public function __construct(
+        private ReadContactGroupRepositoryInterface $repository,
+        private ContactInterface $user
+    ) {
+    }
+
+    /**
+     * @param FindContactGroupsPresenterInterface $presenter
+     */
+    public function __invoke(FindContactGroupsPresenterInterface $presenter): void
+    {
+        try {
+            if ($this->user->isAdmin()) {
+                $contactGroups = $this->repository->findAll();
```

Comment:
Simple remark as we are currently in the ContactGroup context, is it necessary to define `contactGroups` ?
Could it be only `groups` ?

Reply:
I think it's more readable to have contactGroups, you don't have to remember in what context you are (even if it's indicate in the class name few lines above)

Reply:
Seen with @jeremyjaouen let's keep it that way